lastgersync/processor.go

```diff
@@ -74,17 +85,28 @@ func (p *processor) ProcessBlock(ctx context.Context, block sync.Block) error {
 	if err := meddler.Insert(tx, "block", &BlockNum{Num: block.Num}); err != nil {
 		return err
 	}
-	for _, e := range block.Events {
-		event, ok := e.(Event)
+
+	for _, genericEvt := range block.Events {
+		event, ok := genericEvt.(*Event)
 		if !ok {
-			return errors.New("failed to convert sync.Block.Event to Event")
+			return fmt.Errorf("unexpected event type %T", event)
```

Comment:
```suggestion
			return fmt.Errorf("unexpected event type %T", genericEvt)
```
shouldn't `event` will be nil if cast fails ?